src/backend/api/reservations.js

```diff
@@ -0,0 +1,67 @@
+const express = require('express');
+const router = express.Router();
+const knex = require('../database');
+
+router.get('/', async (req, res) => {
+  try {
+    const result = await knex.from('reservation').select();
+    res.json(result);
+  } catch (error) {
+    res.status(500).json({ Msg: 'Internal server error' });
+  }
+});
+
+router.post('/', async (req, res) => {
+  try {
+    const result = await knex.insert(req.body).into('reservation');
+    res.status(201).send('Created');
+  } catch (error) {
+    res.status(500).json({ Msg: 'Internal server error' });
+  }
+});
+
+router.get('/:id', async (req, res) => {
+  try {
+    const paramsValue = req.params.id;
+    const result = await knex('reservation').where({ id: paramsValue });
+    res.send(result);
+  } catch (error) {
+    res.status(500).json({ Msg: 'Internal server error' });
+  }
+});
+
+router.put('/:id', async (req, res) => {
+  try {
+    const paramsValue = req.params.id;
+    const result = await knex('reservation')
+      .where({ id: req.params.id })
+      .update({ contact_name: 'Vie dan Damayanti' });
+    return res.send('Data is updated');
+  } catch (error) {
+    res.status(500).json({ Msg: 'Internal server error' });
+  }
+});
+
+router.delete('/:id', async (req, res) => {
+  try {
+    const paramsValue = req.params.id;
+    const result = await knex('reservation').where({ id: paramsValue }).del();
+    return res.send('Reservation is deleted');
+  } catch (error) {
+    res.status(500).json({ Msg: 'Internal server error' });
+  }
+});
+
+//availableReservations
+router.get('/', async (req, res) => {
+  const reservation = knex('reservation');
+  const isAvailable = Number(req.query.availableReservations === '1');
+  if (isAvailable) {
+    const result = await reservation.where('availability', '=', isAvailable);
```

Comment:
You should try joining the meals table to compare the availability of the reservations.
But You have already created a query for this in the meals.js!!